service/src/test/java/bio/terra/tanagra/plugin/PluginTest.java

```diff
@@ -0,0 +1,24 @@
+package bio.terra.tanagra.plugin;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.tanagra.service.accesscontrol.AccessControlPlugin;
+import bio.terra.tanagra.service.accesscontrol.OpenAccessControlPlugin;
+import bio.terra.tanagra.testing.BaseSpringUnitTest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ActiveProfiles;
+
+@ActiveProfiles("test")
+public class PluginTest extends BaseSpringUnitTest {
+  private static final String TEST_UNDERLAY = "test";
+
+  @Autowired private PluginService pluginService;
+
+  @Test
+  void defaultPluginLoading() {
+    AccessControlPlugin accessControlPlugin =
+        pluginService.getPlugin(TEST_UNDERLAY, AccessControlPlugin.class);
+    assertTrue(accessControlPlugin instanceof OpenAccessControlPlugin);
```

Comment:
Should this be `instanceof TestPluginExternalImplementation`? In your test underlay I see that as the plugin implementation class name.

Not sure if this is what you were going for, but perhaps you can test your plugin loading with 2 tests:
- No plugin specified, check it loads the default.
- TestImplementationClass specified, check that it throws a "class not found" or some other exception (confirming that it tried to load the non-default).
- (When we have another actual implementation class, not in this PR) RealImplementationClass specified, check that it loads and that the class name matches.

Reply:
No, I had to remove the `TEST` plugin type from the `PluginTest` enum, so I can only use "real" plugins for testing.

I missed updating the plugin configuration in the underlay when I removed the test code. Thanks for catching.